### PR TITLE
VirusTotal - fixed invalid URLs handling

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32785

## Description
Currently, invalid URLs are being sent to VT. Added a fix for that using the integration's URL regex.

## Screenshots
Using this URLs list:
image

Before the fix:
image

After the fix:
image

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No
   Well, currently sending `demisto.com` to the URL command will return results. but that is not correct since its not a URL.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml [hunk 1]
```diff
@@ -520,34 +520,27 @@ script:
...
-                }
-                z = i;
-            } else {
+                nonUrls.push(urlList[z])
+            }
+            else {
                 urls.push(urlList[z]);
             }
         }
+        return [urls, nonUrls];
```
Comment (yaakovi) [inline Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml:534]:
```suggestion
        return {validUrls: urls, invalidUrls: nonUrls};
```

File: Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml [hunk 2]
```diff
@@ -520,34 +520,27 @@ script:
...
 
-        return urls;
     }
     function doURL(url, threshold, longFormat, sampleSize, submitWait, waitForRateLimit, retries) {
         //lowercase the URL protocol
         //Example: https://www.demisto.com --> https://www.demisto.com, Http://www.demisto.com --> http://www.demisto.com, www.demisto.com --> www.demisto.com
-
-        var urlList = getURLs(url);
+        var urls = getURLs(url);
+        var urlList = urls[0];
```
Comment (yaakovi) [inline Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml:541]:
```suggestion
        var urlList = urls.validUrls;
```

File: Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml [hunk 3]
```diff
@@ -662,6 +655,18 @@ script:
                 EntryContext: ec
             });
         }
+
+        nonUrls = urls[1]
```
Comment (yaakovi) [inline Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml:659]:
```suggestion
        nonUrls = urls.invalidUrls
```

File: Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml [hunk 4]
```diff
@@ -520,34 +520,27 @@ script:
...
-                while(i>=0 && z > 0 && optionalURL == null) {
-                    i--;
-                    optionalURL = urlList.slice(i, z).join("").match(/(?:(?:https?|ftp|hxxps?):\/\/|www\[?\.\]?|ftp\[?\.\]?)(?:[-\w\d]+\[?\.\]?)+[-\w\d]+(?::\d+)?(?:(?:\/|\?)[-\w\d+&@#\/%=~_$?!\-:,.\(\);]*[\w\d+&@#\/%=~_$\(\);])?/);
-                }
-                if (optionalURL !== null) {
-                    urls.push(urlList.slice(i, z+1).join(","));
-                }
-                z = i;
-            } else {
+                nonUrls.push(urlList[z])
```
Comment (yaakovi) [inline Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml:528]: why did you remove the URL manipulation code?

**Review by yaakovi — CHANGES_REQUESTED** [review]

File: Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml (hunk 4 above)
Comment (reutshal) [reply Packs/VirusTotal/Integrations/integration-VirusTotal_5.5.yml:528]: This code was going over the rest of the list and trying to match the same regex from above, and caused bad URLs to be added to the list.
For example:
On this list - 
image
The integration now returns an error- Invalid URL.
A deeper look will show it's trying to send the string 'https://www.google.com,demisto.com,google.com' as one URL to the service.
So it's a bug + I couldn't understand this logic. 
Why not just try to match each of the URLs in the list to the regex?

File: Packs/VirusTotal/pack_metadata.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "name": "VirusTotal",
     "description": "Analyze suspicious hashes, URLs, domains and IP addresses",
     "support": "xsoar",
-    "currentVersion": "1.0.2",
+    "currentVersion": "1.1.0",
```
[the remark this answers is not in the document]
Comment (reutshal) [reply Packs/VirusTotal/pack_metadata.json:5]: Will downgrade to revision

**Review by yaakovi — APPROVED** [review]

**Review by bakatzir — APPROVED** [review]
